### ISSUE TITLE
Cannot test set equality

### ISSUE THREAD

**Issue by russel** [renatoathaydes/specks#12]
In a property test I have the assertion function:

```
(Set<String> s, Set<String> w) =>expect(s, equalTo(w))
```

Sadly though this causes the error:

```
Argument must be assignable to parameter matcher of expect: Matcher<Set<String>&Comparable<Set<String>>> is not assignable to Matcher<Set<String>>
```

and I have no idea where to go to fix this. Hopefully this is just me not knowing.

**Comment by renatoathaydes** [comment]
This code also does not compile:

``` ceylon
Set<String> s1 = set { "a", "B" };
Set<String> s2 = set { "a", "b" };
print(s1 <=> s2);
```

Because Sets are not Comparable. You can use `sameAs` for non-Comparable equal comparisons:

``` ceylon
test
shared Specification setsAreEqual() => Specification {
    expectations {
        expect(set { 1, 2, 3 }, sameAs(set { 1, 2, 3 }))
    }
};
```

I've been thinking of making `equalTo` behave as `sameAs` as I see no advantage in having two separate assertions (as you probably remember now, `equalTo` uses `<=>` for comparison, while `sameAs` uses `==` and also works with null).

**Comment by russel** [comment]
Curses, I had forgotten all about that, this is a rehash of a conversation we have already had. My fault for forgetting, thanks for your patience in explaining.

Is there anything else I have forgotten before I do the Ceylon (and Kotlin) mini-workshop on Saturday at /dev/summer? :-)

**Comment by renatoathaydes** [comment]
No, but please don't say how much specks sucks because of this ;) we're still before 1.0 so we can fix these things! As I said, I plan to make `equalTo` work just like `sameAs` and maybe remove `sameAs`. What do you say?

**Comment by russel** [comment]
I can live with using sameAs rather than equalTo, the problem is remembering the rules. I think that if the reflection is possible so as to use <=> if it exists and == if it doesn't then that would be a good thing.

Specks is the only sane test framework for Ceylon.
